### ISSUE TITLE
Problems about visualizing Oakink-Image dataset

### ISSUE THREAD

**Issue by Sherlock1956** [lixiny/OakInk#14]
Thank you for the amazing work on the OakInk dataset! It is a tremendous resource for the community, and I truly appreciate all the effort that has gone into making it so comprehensive and accessible.

I am currently trying to visualize the OakInk-Image dataset using the MANO layer implementation from this [repository](https://github.com/otaheri/MANO). While the code works without any errors, I have encountered an issue where the generated hand and the object do not seem to align correctly in the visualization as the picture I put below. I have reviewed the dataset documentation and my code multiple times, but I am unable to identify where the problem lies.

Below is the code I am using for visualization:
```
import torch
import mano
from mano.utils import Mesh
import pickle
from scipy.spatial.transform import Rotation as R
import numpy as np
import trimesh
general_info = "/Users/lyx/Downloads/dataset/OkaInk/anno/general_info/A01001_0001_0000__2021-09-26-19-59-58__0__16__0.pkl"
obj_path = "/Users/lyx/Downloads/dataset/OkaInk/obj/A01001.obj"

with open(general_info, "rb") as file: 
    data = pickle.load(file)

# load object and apply transformation
obj_anno = data["obj_anno"]
obj_mesh = trimesh.load(obj_path)
obj_mesh.apply_transform(obj_anno)

# cam_extr = data["cam_extr"]
betas = data["hand_anno"]["hand_shape"].unsqueeze(0)
transl = data["hand_anno"]["hand_tsl"].unsqueeze(0)

rotation = R.from_quat(data["hand_anno"]["hand_pose"][:, [1, 2, 3, 0]])
axis_angle = rotation.as_rotvec().astype(np.float32)  
global_orient = torch.from_numpy(axis_angle.flatten()[0:3]).unsqueeze(0)
pose = torch.from_numpy(axis_angle.flatten()[3:]).unsqueeze(0)

model_path = '/Users/lyx/Downloads/dataset/OkaInk/MANO/models'
n_comps = 45
batch_size = 1
rh_model = mano.load(model_path=model_path,
                     is_rhand= True,
                     num_pca_comps=n_comps,
                     batch_size=batch_size,
                     flat_hand_mean=False)

output = rh_model(betas=betas,
                  global_orient=global_orient,
                  hand_pose=pose,
                  transl=transl,
                  return_verts=True,
                  return_tips = True)

h_meshes = rh_model.hand_meshes(output)
j_meshes = rh_model.joint_meshes(output)

#visualize hand and joint and object meshes
hj_meshes = Mesh.concatenate_meshes([h_meshes[0], j_meshes[0]])
hjo_meshe = Mesh.concatenate_meshes([obj_mesh,hj_meshes])
hjo_meshe.show() 
```
I would be very grateful if you could provide any guidance or point out potential mistakes I might have overlooked. Thank you again for this incredible dataset and for making it available to the research community!
<img width="890" alt="image" src="https://github.com/user-attachments/assets/41705f23-bc0e-4d34-8fcf-b861a150596a">

**Comment by Sherlock1956** [comment]
> Thank you for the amazing work on the OakInk dataset! It is a tremendous resource for the community, and I truly appreciate all the effort that has gone into making it so comprehensive and accessible.
> 
> 
> 
> I am currently trying to visualize the OakInk-Image dataset using the MANO layer implementation from this [repository](https://github.com/otaheri/MANO). While the code works without any errors, I have encountered an issue where the generated hand and the object do not seem to align correctly in the visualization as the picture I put below. I have reviewed the dataset documentation and my code multiple times, but I am unable to identify where the problem lies.
> 
> 
> 
> Below is the code I am using for visualization:
> 
> ```
> 
> import torch
> 
> import mano
> 
> from mano.utils import Mesh
> 
> import pickle
> 
> from scipy.spatial.transform import Rotation as R
> 
> import numpy as np
> 
> import trimesh
> 
> general_info = "/Users/lyx/Downloads/dataset/OkaInk/anno/general_info/A01001_0001_0000__2021-09-26-19-59-58__0__16__0.pkl"
> 
> obj_path = "/Users/lyx/Downloads/dataset/OkaInk/obj/A01001.obj"
> 
> 
> 
> with open(general_info, "rb") as file: 
> 
>     data = pickle.load(file)
> 
> 
> 
> # load object and apply transformation
> 
> obj_anno = data["obj_anno"]
> 
> obj_mesh = trimesh.load(obj_path)
> 
> obj_mesh.apply_transform(obj_anno)
> 
> 
> 
> # cam_extr = data["cam_extr"]
> 
> betas = data["hand_anno"]["hand_shape"].unsqueeze(0)
> 
> transl = data["hand_anno"]["hand_tsl"].unsqueeze(0)
> 
> 
> 
> rotation = R.from_quat(data["hand_anno"]["hand_pose"][:, [1, 2, 3, 0]])
> 
> axis_angle = rotation.as_rotvec().astype(np.float32)  
> 
> global_orient = torch.from_numpy(axis_angle.flatten()[0:3]).unsqueeze(0)
> 
> pose = torch.from_numpy(axis_angle.flatten()[3:]).unsqueeze(0)
> 
> 
> 
> model_path = '/Users/lyx/Downloads/dataset/OkaInk/MANO/models'
> 
> n_comps = 45
> 
> batch_size = 1
> 
> rh_model = mano.load(model_path=model_path,
> 
>                      is_rhand= True,
> 
>                      num_pca_comps=n_comps,
> 
>                      batch_size=batch_size,
> 
>                      flat_hand_mean=False)
> 
> 
> 
> output = rh_model(betas=betas,
> 
>                   global_orient=global_orient,
> 
>                   hand_pose=pose,
> 
>                   transl=transl,
> 
>                   return_verts=True,
> 
>                   return_tips = True)
> 
> 
> 
> h_meshes = rh_model.hand_meshes(output)
> 
> j_meshes = rh_model.joint_meshes(output)
> 
> 
> 
> #visualize hand and joint and object meshes
> 
> hj_meshes = Mesh.concatenate_meshes([h_meshes[0], j_meshes[0]])
> 
> hjo_meshe = Mesh.concatenate_meshes([obj_mesh,hj_meshes])
> 
> hjo_meshe.show() 
> 
> ```
> 
> I would be very grateful if you could provide any guidance or point out potential mistakes I might have overlooked. Thank you again for this incredible dataset and for making it available to the research community!
> 
> <img width="890" alt="image" src="https://github.com/user-attachments/assets/41705f23-bc0e-4d34-8fcf-b861a150596a">
> 
> 

I discover that the hand pose in the picture seems to be incorrect, but the relative position seems right. Am I use the mano incorrectly? I had little problems forming the axis-angle from the dataset and I don't know whether if I do it right

**Comment by lixiny** [comment]
try to use this script to visualize: 
https://github.com/oakink/OakInk/blob/main/scripts/viz_oakink_image_seq.py. 

e.g:
```
python scripts/viz_oakink_image_seq.py --draw_mode mesh --view_id 0
```

I got: 
![截屏2024-11-24 14 00 01](https://github.com/user-attachments/assets/641b4322-356a-490c-95f9-234c3d5810d4)

seems the results is correct.

**Comment by Sherlock1956** [comment]
@lixiny Thank you for your prompt response and for providing the script! I appreciate your help and will try it out.
